helm-charts/konk/templates/init.yaml

```diff
@@ -0,0 +1,61 @@
+apiVersion: apps/v1
+kind: Deployment
+metadata:
+  name: {{ include "konk.fullname" . }}-init
+  namespace: {{ .Release.Namespace }}
+  labels:
+    {{- include "konk.labels" . | nindent 4 }}
+spec:
+  replicas: 1
+  selector:
+    matchLabels:
+      {{- include "konk.selectorLabels" . | nindent 6 }}
+      app.kubernetes.io/component: init
+  template:
+    metadata:
+      labels:
+        {{- include "konk.selectorLabels" . | nindent 8 }}
+        app.kubernetes.io/component: init
+    spec:
+      serviceAccountName: {{ include "konk.serviceAccountName" . }}
+      initContainers:
```

Comment:
Just to clarify, since this is a separate deployment, the `initContainer` approach will work since the secrets being created here are mounted to the pod created by the other deployment. Correct?

Reply:
Yes, this deployment/pod has no dependency on the secrets it creates. The other deployment does have a dependency on these secrets, which is why the provision.sh cannot be in that other deployment.